packages/block-library/src/cover/transforms.js

```diff
@@ -51,6 +51,55 @@ const transforms = {
 					]
 				),
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/group' ],
+			isMatch: ( { backgroundColor, gradient, style } ) => {
+				/*
+				 * Make sure the group has a background, since if the cover block is
+				 * created without one, it displays the "placeholder" state,
+				 * which makes it look like it has no contents.
+				 */
+				return (
+					backgroundColor ||
+					style?.color?.background ||
+					style?.color?.gradient ||
+					gradient
+				);
+			},
+			transform: (
+				{ align, anchor, backgroundColor, gradient, style },
+				innerBlocks
+			) => {
+				/*
+				 * Clone the blocks to be transformed.
+				 * Failing to create new block references causes the original blocks
+				 * to be replaced in the switchToBlockType call thereby meaning they
+				 * are removed both from their original location and within the
+				 * new cover block.
+				 */
+				const coverInnerBlocks = innerBlocks.map( ( block ) => {
+					return createBlock(
+						block.name,
+						block.attributes,
+						block.innerBlocks
+					);
+				} );
+
+				return createBlock(
+					'core/cover',
+					{
+						align,
+						anchor,
+						overlayColor: backgroundColor,
+						customOverlayColor: style?.color?.background,
+						gradient,
+						customGradient: style?.color?.gradient,
+					},
+					coverInnerBlocks
```

Comment:
You should use `createBlocksFromInnerBlocksTemplate` from `blocks API` here, that recursively creates new blocks from innerBlocks.
``` createBlocksFromInnerBlocksTemplate( innerBlocks ) ```

`coverInnerBlocks` above in your code keeps the same reference to innerBlocks so it doesn't handle the case where an inner block has more inner blocks itself.

Reply:
Hey 👋 - I made the suggestion by seeing the comment and not investigating more... In general I have run into the `unique` block client ids problems in the past and sounded like this was the case as well.

After pointing out the existing code, I'll need to investigate a bit more why we do this in `group` (from before) and what is really needed. FYI when tested your PR before `createBlocksFromInnerBlocksTemplate` suggestion, there was no error/crash but the innerBlock client ids were the same - thus my comment 😄 

Thanks again for your work here and I'll dig deeper really soon.

Reply:
Of course! Sounds good; thanks so much!